### PR TITLE
Update toast notification design

### PR DESCRIPTION
**Pull request by baomastr**

### CODE REVIEW

File: src/clients/web3/useAuth/index.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
   UserRejectedRequestError as UserRejectedRequestErrorWalletConnect,
   WalletConnectConnector,
 } from '@web3-react/walletconnect-connector';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/clients/web3/useAuth/index.ts]: Outside of the component directory it should be imported directly from components
import { toast } from 'components';

When we deprecate the other components and get rid of the v2 directory the only paths we should have to change are the `index.ts` in components

File: src/components/v2/Notice/index.tsx [hunk 2]
```diff
@@ -29,22 +29,30 @@ export const Notice = ({ className, title, description, variant = 'info' }: INot
...
+          {title && (
+            <Typography variant="small2" color="text.primary" css={styles.title}>
+              {title}
+            </Typography>
+          )}
+          {description && (
+            <Typography
+              variant="small2"
+              color="text.primary"
+              css={styles.getDescription({ hasMarginTop: Boolean(title) })}
```
Comment (coreyar) [inline src/components/v2/Notice/index.tsx]: nit: for consistency we should cast booleans the same way I think `!!` is nice shorthand for doing this.

File: src/components/v2/Toast/index.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+const CloseButton = ({ closeToast }: { closeToast: () => void }) => {
+  const classes = useStyles();
+  return (
+    <Button css={classes.btnClose} onClick={closeToast} variant="text">
+      <Icon name="close" />
+    </Button>
+  );
+};
+
+const ToastComponent = ({ title, description, type = 'success' }: ToastProps) => {
```
Comment (coreyar) [inline src/components/v2/Toast/index.tsx]: Type as React functional component `React.FC<IToastProps>`

File: src/components/v2/Toast/index.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import { Button } from '../Button';
+
+import 'react-toastify/dist/ReactToastify.min.css';
+import { customToastGlobalStyles, useStyles } from './styles';
+
+interface ToastArgs {
+  title: string;
+  description?: string;
+}
+interface ToastProps extends ToastArgs {
```
Comment (coreyar) [inline src/components/v2/Toast/index.tsx]: We are using a convention of prefacing interfaces with I

File: src/components/v2/Toast/index.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { GlobalStyles } from '@mui/material';
+
+import { Notice } from '../Notice';
+import { Icon } from '../Icon';
+import { Button } from '../Button';
+
+import 'react-toastify/dist/ReactToastify.min.css';
+import { customToastGlobalStyles, useStyles } from './styles';
+
+interface ToastArgs {
```
Comment (coreyar) [inline src/components/v2/Toast/index.tsx]: We are using a convention of prefacing interfaces with I

File: src/components/v2/Toast/index.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  title: string;
+  description?: string;
+}
+interface ToastProps extends ToastArgs {
+  type: 'success' | 'info' | 'error';
+}
+
+const CloseButton = ({ closeToast }: { closeToast: () => void }) => {
```
Comment (coreyar) [inline src/components/v2/Toast/index.tsx]: Type as React functional component React.FC<{ closeToast: () => void }>

File: src/containers/Main/Faucet.tsx [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { withRouter, RouteComponentProps } from 'react-router-dom';
 import { useRequestFaucetFunds } from 'clients/api';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/containers/Main/Faucet.tsx]:
```suggestion
import { toast } from 'components';
```

File: src/containers/Main/VoteOverview.tsx [hunk 8]
```diff
@@ -14,7 +14,7 @@ import VoteCard from 'components/Vote/VoteOverview/VoteCard';
 import ProposalDetail from 'components/Vote/VoteOverview/ProposalDetail';
 import ProposalHistory from 'components/Vote/VoteOverview/ProposalHistory';
 import { promisify } from 'utilities';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/containers/Main/VoteOverview.tsx]:
```suggestion
import { toast } from 'components';

```

File: src/context/AuthContext.tsx [hunk 9]
```diff
@@ -3,7 +3,7 @@ import noop from 'noop-ts';
 import copyToClipboard from 'copy-to-clipboard';
 
 import { Connector, useAuth } from 'clients/web3';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/context/AuthContext.tsx]:
```suggestion
import { toast } from 'components';
```

File: src/pages/ConvertVrt/Convert/index.tsx [hunk 10]
```diff
@@ -12,7 +12,7 @@ import {
   TokenTextField,
 } from 'components';
 import useSuccessfulTransactionModal from 'hooks/useSuccessfulTransactionModal';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/pages/ConvertVrt/Convert/index.tsx]:
```suggestion
import { toast } from 'components';
```

File: src/pages/ConvertVrt/Withdraw/index.tsx [hunk 11]
```diff
@@ -4,7 +4,7 @@ import BigNumber from 'bignumber.js';
 import { Typography } from '@mui/material';
 import { ConnectWallet, Icon, PrimaryButton } from 'components';
 import useSuccessfulTransactionModal from 'hooks/useSuccessfulTransactionModal';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/pages/ConvertVrt/Withdraw/index.tsx]:
```suggestion
import { toast } from 'components';
```

File: src/pages/Dashboard/Markets/SupplyMarket/index.tsx [hunk 12]
```diff
@@ -3,7 +3,7 @@ import React, { useState } from 'react';
 import { Paper } from '@mui/material';
 import { Asset, TokenId } from 'types';
 import { UiError } from 'utilities/errors';
-import toast from 'components/Basic/Toast';
+import toast from 'components/v2/Toast';
```
Comment (coreyar) [inline src/pages/Dashboard/Markets/SupplyMarket/index.tsx]:
```suggestion
import { toast } from 'components';
```

**Comment by therealemjy** [comment]
Looks good to me. Only one thing: the font family used by the toast content is sans-serif instead of Proxima Nova:
Screen Shot 2022-05-18 at 08 41 28
Screen Shot 2022-05-18 at 08 41 30

File: src/components/v2/Toast/index.tsx [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+import { Notice } from '../Notice';
+import { Icon } from '../Icon';
+import { Button } from '../Button';
+
+import 'react-toastify/dist/ReactToastify.min.css';
+import { customToastGlobalStyles, useStyles } from './styles';
+
+interface IToastArgs {
+  title: string;
+  description?: string;
```
Comment (therealemjy) [inline src/components/v2/Toast/index.tsx]: According to the new designs, notifications don't have titles anymore:
Screen Shot 2022-05-18 at 08 46 48

Comment (baomastr) [reply src/components/v2/Toast/index.tsx]: seems like we have usages of title and description in our components. so I'd leave optional title here.
I'll write to designers slack channel to confirm this

image

Comment (baomastr) [reply src/components/v2/Toast/index.tsx]: removed titles for notice component and for toast notifications

**Comment by baomastr** [comment]
@coreyar  @therealemjy are we ready to merge this PR?

File: src/components/v2/Toast/index.tsx (hunk 13 above)
Comment (therealemjy) [reply src/components/v2/Toast/index.tsx]: Only the notifications don't have titles, the notices still use a title:
Screen Shot 2022-05-19 at 16 42 27

**Comment by therealemjy** [comment]
@baomastr No, the `Notice` component needs to keep its `title`, only it should be optional. The notifications then display using a `description` only.

File: src/components/v2/Toast/index.tsx (hunk 13 above)
Comment (baomastr) [reply src/components/v2/Toast/index.tsx]: done

**Review by therealemjy — APPROVED** [review]

**Pull request merged**